packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -544,22 +614,74 @@ class SliverChildListDelegate extends SliverChildDelegate {
   /// The widgets to display.
   final List<Widget> children;
 
+  final Iterator<Widget> _childIterator;
+
+  /// A map to cache key to index lookup for children
+  ///
+  /// _mapKeyToIndex[null] is used as current index during the lazy loading process
+  /// in [_findChildIndex]. _mapKeyToIndex should never be used for looking up null key.z
+  final Map<Key, int> _mapKeyToIndex;
+
+  bool get _isConstantInstance => _mapKeyToIndex == null;
+
+  int _findChildIndex(Key key){
+    if(_isConstantInstance) {
+      return -1;
+    }
+    // This function should only be called when any child has key during performRebuild.
+    // We do lazy loading in this function.
+    if(!_mapKeyToIndex.containsKey(key)) {
+      int index;
+      while(_childIterator.moveNext()) {
```

Comment:
Tracking index and having an iterator seems to add a lot of complexity. Can we just use a regular for loop that increases indexes?